core/src/main/java/org/elasticsearch/cluster/routing/allocation/NodeRebalanceResult.java

```diff
@@ -73,16 +65,26 @@ public boolean isDeltaAboveThreshold() {
     }
 
     /**
-     * Returns the current weight of the node if the shard is not added to the node.
+     * Returns if the simulated weight of the shard after assigning the node to it would be better than the current node's weight.
+     * If {@code true}, then the shard should be rebalanced to the node, assuming allocation is allowed to the node.
      */
-    public float getCurrentWeight() {
-        return currentWeight;
+    public boolean isBetterWeightWithShardAdded() {
+        return betterWeightWithShardAdded;
     }
 
-    /**
-     * Returns the weight of the node if the shard is added to the node.
-     */
-    public float getWeightWithShardAdded() {
-        return weightWithShardAdded;
+    @Override
+    protected void innerToXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.field("delta_above_threshold", deltaAboveThreshold);
+        if (deltaAboveThreshold) {
+            builder.field("better_weight_with_shard_added", betterWeightWithShardAdded);
```

Comment:
When is the weight delta above threshold but weight not better? I don't understand this conditional output.